File: ios-app/Dictaphone/Dictaphone/Models/AudioRecord.swift
import Foundation

struct AudioRecord: Identifiable, Codable {
    let id: UUID
    let url: URL
    let duration: TimeInterval
    let timestamp: Date
    var transcription: String?
    
    var fileName: String {
        url.lastPathComponent
    }
}

enum RecordingState: Equatable {
    case idle
    case recording
    case playing(progress: Float)
    case error(message: String)
}

struct RecorderState {
    var recordingState: RecordingState = .idle
    var duration: TimeInterval = 0
    var waveformData: [Float] = []
    var isTranscribing: Bool = false
    var error: String? = nil
}

File: android-app/app/src/main/java/com/dictaphone/domain/model/AudioRecord.kt
package com.dictaphone.domain.model

import java.io.File

data class AudioRecord(
    val id: String,
    val file: File,
    val duration: Long,
    val timestamp: Long,
    val transcription: String? = null
)

sealed class RecordingState {
    data object Idle : RecordingState()
    data object Recording : RecordingState()
    data class Playing(val progress: Float = 0f) : RecordingState()
    data class Error(val message: String) : RecordingState()
}

data class RecorderUiState(
    val recordingState: RecordingState = RecordingState.Idle,
    val duration: Long = 0L,
    val waveformData: List<Float> = emptyList(),
    val isTranscribing: Boolean = false,
    val error: String? = null
)
